features/dnsresolver/netty/src/main/java/org/opennms/netmgt/dnsresolver/netty/NettyDnsResolver.java

```diff
@@ -177,24 +190,80 @@ public void destroy() {
 
     @Override
     public CompletableFuture<Optional<InetAddress>> lookup(String hostname) {
-        return circuitBreaker.executeCompletionStage(() -> {
-            final NettyResolverContext resolverContext = iterator.next();
-            final Timer.Context timerContext = lookupTimer.time();
-            return resolverContext.lookup(hostname).whenComplete((res, ex) -> {
-                timerContext.stop();
-            });
-        }).toCompletableFuture();
+        if (lookupEnabled) {
```

Comment:
The DnsResolver is designed to be used a shared service and not solely for flows (even though that's the only thing that's using it right now.)

If we want to disable lookups I'd rather see us avoiding calls to the DnsResolver altogether in the specific components rather than disabling the service globally.

Reply:
I think we need to keep the ability to globally enable/disable functionality and to also do as you suggested, allow resolution functionality to be enable or disabled per flow protocol.

Reply:
I still have some reservations on keeping this flag here. If we want to be able to turn in on/off globally, I'd rather see us load a different feature & implementation of the DnsResolver instead. This would follow a similar pattern to the NullPinger & NoopKvStores.

Reply:
That would streamline it like the other packages, unfortunately it's a bit beyond me at the moment. Would reworking it into a noop package also allow this function to be "enable/disable" on a running system without restarting services? If so I would also have no reservations then.

Speaking as a network administrator, I like the quick nature of enabling or disabling on the fly without disrupting other functionality or restarting services.

Reply:
> Would reworking it into a noop package also allow this function to be "enable/disable" on a running system without restarting services? 

Yep.

Reply:
I looked at NullPinger and NoOpBlobStore and I understand the general idea of both, but I don't think I can implement this myself at my current skill set. Given the current need, can we implement this as a Part 1 and circle back around closing this in Part 2? If so I will update the PR title and make the appopriate notations in the Jira Issue.

Reply:
I'd rather keep this out altogether until we can implement it differently.

Workaround is to disable lookups for all the protocols, or configure the circuit breaker such that it trips quickly.

Reply:
Removed